src/ui.rs

```diff
@@ -257,62 +258,52 @@ fn draw_chat<B: Backend>(f: &mut Frame<B>, app: &mut App, area: Rect) {
     }
 }
 
-fn send_receipts(app: &mut App, height: usize) {
-    let mut timestamps = Vec::new();
+fn prepare_receipts(app: &mut App, height: usize) {
+    let mut to_send = Vec::new();
     let user_id = app.user_id;
-
-    {
-        let channel = app
-            .data
-            .channels
-            .state
-            .selected()
-            .and_then(|idx| app.data.channels.items.get_mut(idx));
-        let channel = match channel {
-            Some(c) if !c.messages.items.is_empty() => c,
-            _ => return,
-        };
-
-        let offset = if let Some(selected) = channel.messages.state.selected() {
-            channel
-                .messages
-                .rendered
-                .offset
-                .min(selected)
-                .max(selected.saturating_sub(height))
-        } else {
-            channel.messages.rendered.offset
-        };
-
-        let messages = &mut channel.messages.items[..];
-
-        let _ = messages
-            .iter_mut()
-            .rev()
-            .skip(offset)
-            .for_each(|msg| match msg.receipt {
-                Receipt::Read => (),
-                _ => {
-                    if msg.from_id != user_id {
-                        timestamps.push(msg.arrived_at);
-                        msg.receipt = Receipt::Read
-                    }
-                }
-            });
-    }
-
     let channel = app
         .data
         .channels
         .state
         .selected()
-        .and_then(|idx| app.data.channels.items.get(idx));
+        .and_then(|idx| app.data.channels.items.get_mut(idx));
     let channel = match channel {
         Some(c) if !c.messages.items.is_empty() => c,
         _ => return,
     };
 
-    app.send_receipts(channel, timestamps, Receipt::Read);
+    let offset = if let Some(selected) = channel.messages.state.selected() {
+        channel
+            .messages
+            .rendered
+            .offset
+            .min(selected)
+            .max(selected.saturating_sub(height))
+    } else {
+        channel.messages.rendered.offset
+    };
+
+    let messages = &mut channel.messages.items[..];
+
+    let _ = messages
+        .iter_mut()
+        .rev()
+        .skip(offset)
+        .for_each(|msg| match msg.receipt {
+            Receipt::Read => (),
+            Receipt::Nothing => (), // Backward-compatibility
+            _ => {
```

Comment:
Maybe it is better to use the name of the variant here? We might add another one, without actually planning to land in this branch.